src/tests/ftest/container/per_server_fault_domain.py

```diff
@@ -0,0 +1,327 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from collections import defaultdict
+from ior_test_base import IorTestBase
+
+
+class PerServerFaultDomainTest(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description:
+    The granularity of redundancy factor can be changed to node (rf_lvl:2). This means
+    that in a two-engine setup, if one of the ranks goes down, we count as one. If both of
+    the ranks in the same node go down, we still count as one. The default mode is engine
+    (rf_lvl:1).
+
+    If the number of ranks that has gone down exceeds the rf value, container status would
+    become UNCLEAN. For example, with an rf:1 container, if two ranks go down, the status
+    would be UNCLEAN. If one rank goes down, it would be HEALTHY.
+
+    Verify the node mode by stopping one/two ranks in one/two nodes.
+
+    Assume that the two ranks in the same node map to the same IP address.
+
+    :avocado: recursive
+    """
+
+    def verify_per_server_fault(self, create_pool, ranks_to_stop, properties,
+                                expected_status):
+        """Run the main test steps.
+
+        1. Create a pool if create_pool is True. Then create a container with given
+        container property.
+        2. Run IOR to write some data to the container.
+        3. Stop the ranks_to_stop ranks.
+        4. Wait for rebuild to finish.
+        5. Get container's health status.
+        6. Restart the stopped servers for cleanup.
+        7. Verify container's Health status is expected.
+
+        Args:
+            create_pool (bool): Whether to create a pool.
+            ranks_to_stop (str): Comma-separated ranks to stop.
+            properties (str): Container property defined in the test yaml (rf_1 or rf_2).
+            expected_status: Expected container status at the end. HEALTHY or UNCLEAN.
+        """
+        # 1. Create a pool if create_pool is True. Then create a container with given
+        # container property.
+        if create_pool:
+            self.pool = self.get_pool()
+        self.container = self.get_container(pool=self.pool, properties=properties)
+
+        # Run IOR to write some data to the container.
+        self.ior_cmd.set_daos_params(
+            group=self.server_group, pool=self.pool, cont_uuid=self.container.uuid)
+        manager = self.get_ior_job_manager_command()
+        self.run_ior(manager=manager, processes=1)
+
+        # 4. Stop the ranks.
+        dmg_command = self.get_dmg_command()
+        dmg_command.system_stop(ranks=ranks_to_stop)
+
+        # 5. Wait for rebuild to finish.
+        self.pool.wait_for_rebuild_to_start(interval=10)
+        self.pool.wait_for_rebuild_to_end(interval=10)
```

Comment:
Out of curiosity, why is the interval increased here?  Is this just to save some job.log length as a trade-off for detecting rebuild quicker?